tests/diffpy/utils/scattering_objects/test_diffraction_objects.py

```diff
@@ -231,94 +231,179 @@ def test_diffraction_objects_equality(inputs1, inputs2, expected):
+        assert str(warn_record[0].message) == (
+            "INFO: no wavelength has been specified. You can continue "
+            "to use the DiffractionObject but some of its powerful features "
+            "will not be available. To specify a wavelength, you can use "
+            "DiffractionObject(wavelength=0.71)."
+        )
+    actual_array = getattr(actual_diffraction_object, function)()
+    return np.allclose(actual_array, expected_array)
+
+
+params_q_to_tth = [
+    # UC1: User specified empty q values (without wavelength)
+    (
+        [None, [], []],
+        [[]],
+    ),
+    # UC2: User specified empty q values (with wavelength)
+    (
+        [4 * np.pi, [], []],
+        [[]],
+    ),
+    # UC3: User specified valid q values (without wavelength)
     # expected tth values are 2*arcsin(q) in degrees
-    actual = DiffractionObject(wavelength=4 * np.pi)
-    setattr(actual, "on_q", [[0, 0.2, 0.4, 0.6, 0.8, 1], [1, 2, 3, 4, 5, 6]])
-    actual_tth = actual.q_to_tth()
-    expected_tth = [0, 23.07392, 47.15636, 73.73980, 106.26020, 180]
-    assert np.allclose(actual_tth, expected_tth)
+    (
+        [None, [0, 0.2, 0.4, 0.6, 0.8, 1], [1, 2, 3, 4, 5, 6]],
+        [[]],
+    ),
+    # UC4: User specified valid q values (with wavelength)
+    # expected tth values are 2*arcsin(q) in degrees
+    (
+        [4 * np.pi, [0, 0.2, 0.4, 0.6, 0.8, 1], [1, 2, 3, 4, 5, 6]],
+        [[0, 23.07392, 47.15636, 73.73980, 106.26020, 180]],
+    ),
+]
+
+
+@pytest.mark.parametrize("inputs, expected", params_q_to_tth)
+def test_q_to_tth(inputs, expected):
+    actual = DiffractionObject(wavelength=inputs[0])
+    actual.on_q = [inputs[1], inputs[2]]
+    expected_tth = expected[0]
+    assert _test_valid_diffraction_objects(actual, "q_to_tth", expected_tth)
 
 
 params_q_to_tth_bad = [
-    # UC1: user did not specify wavelength
+    # UC1: user specified invalid q values that result in tth > 180 degrees
     (
-        [None, [0, 0.2, 0.4, 0.6, 0.8, 1]],
-        "Wavelength is not specified. Please provide a valid wavelength, "
-        "e.g., DiffractionObject(wavelength=0.71).",
+        [4 * np.pi, [0.2, 0.4, 0.6, 0.8, 1, 1.2], [1, 2, 3, 4, 5, 6]],
+        [
+            ValueError,
+            "The supplied q-array and wavelength will result in an impossible two-theta. "
+            "Please check these values and re-instantiate the DiffractionObject.",
+        ],
     ),
-    # UC2: user specified invalid q values that result in tth > 180 degrees
+    # UC2: user specified a wrong wavelength that result in tth > 180 degrees
     (
-        [4 * np.pi, [0.2, 0.4, 0.6, 0.8, 1, 1.2]],
-        "Wavelength * q > 4 * pi. Please check if you entered an incorrect wavelength or q value.",
+        [100, [0, 0.2, 0.4, 0.6, 0.8, 1], [1, 2, 3, 4, 5, 6]],
+        [
+            ValueError,
+            "The supplied q-array and wavelength will result in an impossible two-theta. "
```

Comment:
See above.  Also since we are reusing the error message do we want to minimize word by defining it in a variable once and reusing the variable in multiple tests?

Reply:
Yes I will store them in diffraction_objects.py file